test/parallel/test-node-run.js

```diff
@@ -212,14 +217,76 @@ describe('node --run [command]', () => {
     assert.strictEqual(child.code, 1);
   });
 
-  it('returns error when there is no "scripts" field file', async () => {
+  it('returns error when there is no "scripts" and "bin" fields in file', async () => {
     const child = await common.spawnPromisified(
       process.execPath,
       [ '--run', 'test'],
-      { cwd: fixtures.path('run-script/cannot-find-script') },
+      { cwd: fixtures.path('run-script/cannot-find-script-and-bin') },
     );
-    assert.match(child.stderr, /Can't find "scripts" field in/);
+    assert.match(child.stderr, /Can't find "scripts" or "bin" fields in/);
     assert.strictEqual(child.stdout, '');
     assert.strictEqual(child.code, 1);
   });
+
+  it('print avilables scripts and bins when command not found', async () => {
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run', 'tmp'],
+      { cwd: fixtures.path('run-script') },
+    );
+    assert.match(child.stderr, /Unknown script or bin entry "tmp"/);
+    assert.match(child.stderr, /Available scripts:\n/);
+    assert.match(child.stderr, /ada: ada\n/);
+    assert.match(child.stderr, /Available bins:\n/);
+    assert.match(child.stderr, /bin-test: \.\/test\.js\n/);
```

Comment:
Can you convert this into a snapshot test? It's extremely hard to understand this assertions.